File: Python/1047 - Game Time with minutes.py
StH,StMin,EndH,EndMin = map(int, input().split())
Total_T = [0,0]
if (StH == EndH):
    if (StMin == EndMin):
        Total_T[0] = 24
    if (StMin < EndMin):
        Total_T[1] = (EndMin-StMin)
    if (StMin > EndMin):
        Total_T[0] = 23
        Total_T[1] = ((EndMin+60)-StMin)
elif (StH > EndH):
    if (StMin < EndMin):
        Total_T[0] = ((EndH+24)-StH)
        Total_T[1] = (EndMin-StMin)
    elif (StMin > EndMin):
        Total_T[0] = ((EndH+24)-StH-1)
        Total_T[1] = ((EndMin+60)-StMin)
    else:
        Total_T[0] = ((EndH+24)-StH)
        Total_T[1]=0
else:
    if (StMin < EndMin):
        Total_T[0] = (EndH-StH)
        Total_T[1] = (EndMin-StMin)
    elif (StMin > EndMin):
        Total_T[0] = (EndH-StH-1)
        Total_T[1] = ((EndMin+60)-StMin)
    else:
        Total_T[0] = (EndH-StH)
        Total_T[1]=0
print("O JOGO DUROU %d HORA(S) E %d MINUTO(S)" % (Total_T[0], Total_T[1]))

File: C++/1047 - Game Time with Minutes.cpp
#include<iostream>

void print_result(int hours, int minutes)
{
    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n",hours,minutes);
}

int main()
{
    int start_hour,start_minute,end_hour,end_minute;
    std::cin >> start_hour >> start_minute >> end_hour >> end_minute;
    int minutes,hours;
    if((start_hour == end_hour))
    {
        if (start_minute == end_minute)
        {
            print_result(24,0);
            return 0;
        }
        if (start_minute > end_minute)
        {
            end_minute+=60;
            minutes = end_minute-start_minute;
            print_result(23,minutes);
            return 0;
        }
        else //for readability purposes
        {
            minutes = end_minute-start_minute;
            print_result(0,minutes);
            return 0;
        }
    }
    if(start_hour > end_hour)
    {
        end_hour+= 24;
        hours = end_hour-start_hour;
        if(start_minute == end_minute)
        {
            print_result(hours,0);
            return 0;
        }
        if(start_minute > end_minute)
        {
            end_minute+= 60;
            hours--;
            minutes = end_minute-start_minute;
            print_result(hours,minutes);
            return 0;
        }
        else //for readability purposes
        {
            minutes = end_minute-start_minute;
            print_result(hours,minutes);
            return 0;
        }
    }
    else //for readability purposes
    {
        hours = end_hour-start_hour;
        if(start_minute == end_minute)
        {
            print_result(hours,0);
            return 0;
        }
        if(start_minute > end_minute)
        {
            hours--;
            end_minute+=60;
            minutes = end_minute-start_minute;
            print_result(hours,minutes);
            return 0;
        }
        else //for readability purposes
        {
            minutes = end_minute-start_minute;
            print_result(hours,minutes);
            return 0;
        }
    }
    return 1;
}